### ISSUE TITLE
[regression] hyprctl keyword monitor transform broken (silently ignored)

### ISSUE THREAD

**Issue by ppenguin** [hyprwm/Hyprland#8190]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

***System/Version info***

On NixOS after recent update. Tried versions from current git down to `0.44.0` via flake input, no difference

**EDIT**: I tried `0.43.0` and there screen transform worked. So the regression was introduced after that.

### Description

```
hyprctl keyword monitor 'eDP-1,transform,3'
```
(or other orientations) return ok but don't do anything. Debug log shows nothing suspicious relating.

I've double checked the correct monitor name with `hyprctl monitors`. It also shows the value of `transform: 0` regardless what I applied with `hyprctl`.

### How to reproduce

Do 

```
hyprctl keyword monitor 'eDP-1,transform,3'
```
on recent versions and observe nothing happens.

**Comment by vaxerski** [comment]
can you bisect

**Comment by ppenguin** [comment]
The breakage was introduced with this PR: https://github.com/hyprwm/Hyprland/pull/7571/files
The last _working_ commit is b1ad2d806634edff656cb5ddc9850ae2c73324e8

It was quite tedious to find, because I couldn't reproduce the issue in a nested Hyprland instance...

**Comment by vaxerski** [comment]
are you sure you do not have some output configuration daemon? With that commit, they take priority (kanshi, shikane, etc)

**Comment by ppenguin** [comment]
TBH I was not aware that I was supposed to stop using `kanshi` to keep output config functionality with `hyprctl`, but indeed that was the culprit. I.e. builds after said commit work with `transform` if I don't use `kanshi`.

To handle this issue to completion though, whouldn't we somehow need to handle "the fallout"?
- does the (additional) functionality (more or less?) obsolete tools like `kanshi` or `shikane`
- Is the priority fixed or can it be changed? (Based on specificity or applying attributes that are not explicitly set by the previous handler?) => probably technically not really feasible?
- at least maybe a more prominent mention of **either/or** character of internal vs external display handling (if this is indeed de-facto the case)?

Would it be technically possible to detect the non-success of a `transform` application in `hyprctl`? In that case medium-term it probably could/should be included in the better error handling of `hyprctl` that already is a topic of other issue(s?).

**Comment by vaxerski** [comment]
> Would it be technically possible to detect the non-success of a transform application in hyprctl? In that case medium-term it probably could/should be included in the better error handling of hyprctl that already is a topic of other issue(s?).

The thing is, it doesn't _fail_. It is saved, and when output config gets reloaded it is taken into account. It's just that kanshi set the transform for you, so it takes priority.

> does the (additional) functionality (more or less?) obsolete tools like kanshi or shikane? If so, document in wiki?

what?

> Is the priority fixed or can it be changed? (Based on specificity or applying attributes that are not explicitly set by the previous handler?) => probably technically not really feasible?

Fixed. From most to least important: output manager, dynamic calls, config

**Comment by ppenguin** [comment]
Thanks a lot! That clears it up, I'll close this for now; it appears given this handling it's a matter of smart configuration, at least I know what to look out for. Incidentally, `shikane` didn't seem to inhibit `hyprctl transform` when I tried it ad hoc, so maybe it's more suitable than `kanshi` in that respect.

> > does the (additional) functionality (more or less?) obsolete tools like kanshi or shikane? If so, document in wiki?
> 
> what?

What I meant is that `Hyprland` output management has gotten more elaborate since last time I checked (about a year ago), so possibly I could even do without an output manager now, although I'd have to test the dynamic/hotplug behaviour in more detail.
